crates/ruff_python_formatter/resources/test/fixtures/ruff/expression/named_expr.py

```diff
@@ -11,3 +11,28 @@
 y0 = (y1 := f(x))
 
 f(x:=y, z=True)
+
+assert (x := 1)
+
+
+def f():
+    return (x := 1)
+
+
+for x in (y := [1, 2, 3]):
+    pass
+
+async for x in (y := [1, 2, 3]):
+    pass
+
+del (x := 1)
+
+try:
+    pass
+except (e := Exception):
+    if x := 1:
+        pass
+
+(x := 1)
```

Comment:
I'm trying to figure out why the parentheses _aren't_ being removed here (which is correct, but `parent.is_stmt_expr()` isn't in the conditional).

Reply:
Parentheses are only removed in statements other than `ExprStmt` (and only for top-level expressions). 

https://github.com/astral-sh/ruff/blob/40f54375cbdae6c4c7ae827c5d770b5056f0dde1/crates/ruff_python_formatter/src/statement/stmt_expr.rs#L16-L20

The formatting calls `expr.format()` which preserves parentheses.

(I wonder if we can remove the arithmetic handling now too).